common/taskvalidator/validateworkflow_test.go

```diff
@@ -23,45 +23,102 @@
 package taskvalidator
 
 import (
+	"context"
+	"errors"
 	"testing"
 
 	"github.com/golang/mock/gomock"
 	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
 
 	"github.com/uber/cadence/common/cache"
-	"github.com/uber/cadence/common/log"
 	"github.com/uber/cadence/common/metrics"
+	"github.com/uber/cadence/common/persistence"
 	"github.com/uber/cadence/service/history/constants"
 )
 
+type MockStaleChecker struct {
+	CheckAgeFunc func(response *persistence.GetWorkflowExecutionResponse) (bool, error)
+}
+
+func (m *MockStaleChecker) CheckAge(response *persistence.GetWorkflowExecutionResponse) (bool, error) {
+	return m.CheckAgeFunc(response)
+}
+
 func TestWorkflowCheckforValidation(t *testing.T) {
 	testCases := []struct {
-		name     string
-		domainID string
-		runID    string
+		name          string
+		workflowID    string
+		domainID      string
+		domainName    string
+		runID         string
+		isStale       bool
+		simulateError bool
 	}{
-		{"DomainFetchSuccess", "domain-id-success", "run-id-success"},
-		{"DomainFetchFailure", "domain-id-failure", "run-id-failure"},
+		{"NonStaleWorkflow", "workflow-1", "domain-1", "domain-name-1", "run-1", false, false},
+		{"StaleWorkflow", "workflow-2", "domain-2", "domain-name-2", "run-2", true, false},
+		{"ErrorInGetWorkflowExecution", "workflow-3", "domain-3", "domain-name-3", "run-3", false, true},
 	}
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
 			mockCtrl := gomock.NewController(t)
 			defer mockCtrl.Finish()
 
+			mockLogger := zap.NewNop()
+			mockMetricsClient := metrics.NewNoopMetricsClient()
 			mockDomainCache := cache.NewMockDomainCache(mockCtrl)
-			logger := log.NewNoop()
-			metricsClient := metrics.NewNoopMetricsClient()
-			checker := NewWfChecker(logger, metricsClient, mockDomainCache)
-
-			// Set up the mock behavior for GetDomainByID
+			mockPersistenceRetryer := persistence.NewMockRetryer(mockCtrl)
+			mockStaleChecker := &MockStaleChecker{
+				CheckAgeFunc: func(response *persistence.GetWorkflowExecutionResponse) (bool, error) {
+					return tc.isStale, nil
+				},
+			}
+			checker := NewWfChecker(mockLogger, mockMetricsClient, mockDomainCache, mockPersistenceRetryer, mockStaleChecker)
+			checker.(*checkerImpl).staleCheck = mockStaleChecker
```

Comment:
you already pass `mockStaleChecker` to constructor above. why is this line needed?